collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/IntentGroupTest.java

```diff
@@ -80,17 +92,67 @@ public void appMissingErrorText_ShouldComeFromSpecialFormText() {
+        onView(withId(R.id.play_video)).check(matches(isDisplayed()));
+        onView(withId(R.id.play_video)).check(matches(not(isEnabled())));
+
+        // Check ArbitraryFileWidget without answer
+        onView(withTagValue(is("ArbitraryFileWidgetAnswer"))).check(matches(not(isDisplayed())));
+
         Intent resultIntent = new Intent();
-        resultIntent.putExtra("sometext", "Here is a text value");
 
-        int randomInteger = (new Random()).nextInt(255);
-        resultIntent.putExtra("someinteger", randomInteger);
+        resultIntent.putExtra("questionInteger", "25");
+        resultIntent.putExtra("questionDecimal", "46.74");
+        resultIntent.putExtra("questionText", "sampleAnswer");
 
+        ClipData clipData = ClipData.newRawUri(null, null);
+        clipData.addItem(new ClipData.Item("questionImage", null, createTempFile("famous", "jpg")));
+        clipData.addItem(new ClipData.Item("questionAudio", null, createTempFile("sampleAudio", "wav")));
+        clipData.addItem(new ClipData.Item("questionVideo", null, createTempFile("sampleVideo", "mp4")));
+        clipData.addItem(new ClipData.Item("questionFile", null, createTempFile("fruits", "csv")));
+
+        resultIntent.setClipData(clipData);
         intending(not(isInternal())).respondWith(new Instrumentation.ActivityResult(Activity.RESULT_OK, resultIntent));
         onView(withText("This is buttonText")).perform(click());
 
-        onView(withText("Here is a text value")).check(matches(isDisplayed()));
-        onView(withText(Integer.toString(randomInteger))).check(matches(isDisplayed()));
+        // Check StringWidgets with answers
+        onView(withText("25")).perform(nestedScrollTo()).check(matches(isDisplayed()));
+        onView(withText("46.74")).perform(nestedScrollTo()).check(matches(isDisplayed()));
+        onView(withText("sampleAnswer")).perform(nestedScrollTo()).check(matches(isDisplayed()));
+
+        // Check ImageWidget with answer
+        onView(withTagValue(is("ImageView"))).perform(nestedScrollTo()).check(matches(isDisplayed()));
+        onView(withId(R.id.capture_image)).check(doesNotExist());
+        onView(withId(R.id.choose_image)).check(doesNotExist());
+
+        // Check AudioWidget with answer
+        onView(withId(R.id.audio_controller)).perform(nestedScrollTo()).check(matches(isDisplayed()));
+
+        // Check VideoWidget with answer
+        onView(withId(R.id.play_video)).perform(nestedScrollTo()).check(matches(isDisplayed()));
+        onView(withId(R.id.play_video)).check(matches(isEnabled()));
+
+        // Check ArbitraryFileWidget with answer
+        onView(withTagValue(is("ArbitraryFileWidgetAnswer"))).perform(nestedScrollTo()).check(matches(isDisplayed()));
+    }
+
+    @SuppressWarnings("PMD.DoNotHardCodeSDCard")
+    private Uri createTempFile(String name, String extension) throws IOException {
+        File file = File.createTempFile(name, extension, new File("/sdcard"));
```

Comment:
Out of interest did you need the parent (`new File("/sdcard")`) here? I was under the impression we could just create a file outright and Android handles it.

Reply:
Without it I get `java.io.IOException: Read-only file system`

Reply:
Huh strange. The same thing just worked™️ for me here: https://github.com/getodk/collect/pull/4190/files#diff-83ee5909b031fabf1a76172164745099e708b293a264d0ae3f2cf9dce844ae21R154.

Reply:
Ah ok this way it could work but then I need to use that file via fileprovider that means I would need to edit our `provider_paths.xml` file and add `<cache-path path="/" name="files" />` there. Since we would need that only for tests I would need to create such a separate file for tests only. Is it worth the effort?